### PR TITLE
Copy properties defaults on node creation

### PR DESCRIPTION
**Pull request by marthacryan**
Copies properties defaults into `app_data` field of nodes on node creation. Fixes elyra-ai/elyra#1785
 
Developer's Certificate of Origin 1.1

       By making a contribution to this project, I certify that:

       (a) The contribution was created in whole or in part by me and I
           have the right to submit it under the Apache License 2.0; or

       (b) The contribution is based upon previous work that, to the best
           of my knowledge, is covered under an appropriate open source
           license and I have the right under that license to submit that
           work with modifications, whether created in whole or in part
           by me, under the same open source license (unless I am
           permitted to submit under a different license), as indicated
           in the file; or

       (c) The contribution was provided directly to me by some other
           person who certified (a), (b) or (c) and I have not modified
           it.

       (d) I understand and agree that this project and the contribution
           are public and that a record of the contribution (including all
           personal information I submit with it, including my sign-off) is
           maintained indefinitely and may be redistributed consistent with
           this project or the open source license(s) involved.

### CODE REVIEW

**Comment by bourdakos1** [comment]
I think this looks okay so far, but I need to spend some more time testing and wrapping my head around this, before approving

File: packages/pipeline-editor/src/PipelineController/index.ts [hunk 1]
```diff
@@ -130,13 +130,18 @@ class PipelineController extends CanvasController {
       offsetY: item.y ?? 40,
       nodeTemplate: this.convertNodeTemplate(nodeTemplate),
```
Comment (bourdakos1) [inline packages/pipeline-editor/src/PipelineController/index.ts:131]: we also pass a pipelineId sometimes

**Review by bourdakos1 — CHANGES_REQUESTED** [review]
Since the "defaults" are now just considered "initial" we should also update any of the property resolution logic (tooltip, validate, etc..) to only pull from the node app_data and ignore defaults

**Comment by marthacryan** [comment]
@bourdakos1 Just removed the tooltip / validation code that uses `current_parameters` as default - do you know of any other places that use it?

**Review by bourdakos1 — CHANGES_REQUESTED** [review]
looks good, last thing I noticed is there is also a `newFileNode` action that we might want to also merge into the `addNode` pipeline controller too

**Comment by bourdakos1** [comment]
> Just removed the tooltip / validation code that uses current_parameters as default - do you know of any other places that use it?

@marthacryan I think that's it

File: packages/pipeline-services/src/validation/index.test.ts [hunk 2]
```diff
@@ -146,65 +146,6 @@ describe("getNodeProblems", () => {
     expect(problems[0].info.property).toBe("filename");
   });
 
-  it("should not have issues when required property has a default value", () => {
```
Comment (bourdakos1) [inline packages/pipeline-services/src/validation/index.test.ts:149]: I think we should update this test case to mark it as an error if the pipeline has an "initial" value and then the value is deleted, it should have a validation error if required

**Review by bourdakos1 — CHANGES_REQUESTED** [review]

File: packages/pipeline-services/src/validation/index.test.ts (hunk 2 above)
Comment (marthacryan) [reply packages/pipeline-services/src/validation/index.test.ts:149]: Does what I just pushed look okay?

File: packages/pipeline-services/src/validation/index.test.ts [hunk 3]
```diff
@@ -202,7 +202,7 @@ describe("getNodeProblems", () => {
     };
 
     const problems = getNodeProblems(pipeline, [nodeSpec]);
-    expect(problems).toHaveLength(0);
+    expect(problems).toHaveLength(1);
```
Comment (bourdakos1) [inline packages/pipeline-services/src/validation/index.test.ts:205]:
```suggestion
    expect(problems).toHaveLength(1);
    expect(problems[0].info.property).toBe("has_default");
```

**Review by bourdakos1 — CHANGES_REQUESTED** [review]

File: packages/pipeline-services/src/validation/index.test.ts (hunk 3 above)
Comment (bourdakos1) [reply packages/pipeline-services/src/validation/index.test.ts:205]: we should check what the error is instead of just the length

**Review by bourdakos1 — APPROVED** [review]
lgtm pending ci

**Pull request merged**